lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/tasklet/DefaultTaskletExecutor.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver.tasklet;
+
+import org.apache.reef.driver.context.ActiveContext;
+import org.apache.reef.vortex.common.CommandExecuteTasklet;
+import org.apache.reef.vortex.driver.CommandSender;
+
+import javax.inject.Inject;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The wrapper class around ActiveContext for our convenience.
+ */
+public class DefaultTaskletExecutor implements TaskletExecutor {
+  private final CommandSender commandSender;
+  private final ActiveContext activeContext;
+  private final ConcurrentHashMap<Integer, Tasklet> runningTasklets = new ConcurrentHashMap<>();
+
+  @Inject
+  public DefaultTaskletExecutor(final CommandSender commandSender, final ActiveContext activeContext) {
+    this.commandSender = commandSender;
+    this.activeContext = activeContext;
+  }
+
+  @Override
+  public <TInput extends Serializable, TOutput extends Serializable>
+      void execute(final Tasklet<TInput, TOutput> tasklet) {
+    runningTasklets.put(tasklet.getId(), tasklet);
```

Comment:
Yes we do guarantee uniqueness of id, since it purpose is to distinguish a tasklet from others. I'll see if we can add some checkers to throw an exception if this constraint is founded to be violated.